src/test/test-execute.c

```diff
@@ -1299,13 +1299,16 @@ static int prepare_ns(const char *process_name) {
                 assert_se(mkdir_p(PRIVATE_UNIT_DIR, 0755) >= 0);
 
                 /* Mount tmpfs on the following directories to make not StateDirectory= or friends disturb the host. */
-                FOREACH_STRING(p, "/dev/shm", "/root", "/tmp", "/var/tmp", "/var/lib", PRIVATE_UNIT_DIR)
+                FOREACH_STRING(p, "/dev/shm", "/tmp", "/var/tmp", "/var/lib", PRIVATE_UNIT_DIR)
```

Comment:
Instead of delaying /root, maybe we should first mount PRIVATE_UNIT_DIR, copy data, then overmount tmpfs on /dev/shm, /root, /tmp, /var/tmp, /var/lib ? Then, the build directory can be /tmp or so.

Reply:
Makes sense, updated.